### DISCUSSION TITLE
Print `url` of an issue or list of issues

### DISCUSSION THREAD

**Discussion by mohit2152sharma** [ankitpokhrel/jira-cli#625] [Q&A] [answered]
I can view my issues with the following command `jira issue list -a$(jira me)`. I want to get the link (url) of each issue programmatically. I know that information is available somewhere as I can open the link of an issue in browser when I hit `Enter`. Just wanted to know, what would be the way to extract that `url`, so that I can use the `url` in other shell functions.

**Comment by olivergondza** [comment]
Any help is appreciated, as I tried several things, but to no avail.

- `jira issue list` seems to have no column for it.
- I do not see a way to get `jira issue view` to print the URL.
- `jira serverinfo` does not seem to have a way to print the server prefix.

Any help appreciated.

**Comment by ankitpokhrel** [comment]
Hi @mohit2152sharma @olivergondza, its not possible to get `url` from the list command. However building the issue url is pretty straightforward (example script below).

```sh
#!/usr/bin/env bash

HOST=https://jira.server
TICKETS=$(jira issue list --plain --no-headers --columns=KEY)

for ticket in ${TICKETS}
do
  echo "${HOST}/browse/${ticket}"
done
```

Does this work for your usecase?

**Reply by mohit2152sharma** [reply]
yeah, I figured and doing something similar at the moment. But I thought (and hoped) that there was some command to get the url, as it is already available in some contexts. For example, if I open a list of issues in the viewer, there is a command to copy the issue url to clipboard. 
Is this something planned for future?

**Reply by ankitpokhrel** [reply, marked as answer]
There is an `open` command which I completely forgot about even though I almost use it daily 😬 

```sh
$ jira open --help
Open opens issue in a browser. If the issue key is not given, it will open the project page.

$ jira open ISS-123 --no-browser
https://jira.server/browse/ISS-123
```

**Reply by mohit2152sharma** [reply]
Thanks this is exactly what I was looking for.

**Reply by olivergondza** [reply]
Ah, have not thought of that. Thanks!
